lib/pact/pact_broker/fetch_pacts.rb

```diff
@@ -4,51 +4,53 @@
 module Pact
   module PactBroker
     class FetchPacts
-      attr_reader :provider, :tags, :broker_base_url, :basic_auth_options, :http_client, :index_entity, :all_pacts
+      attr_reader :provider, :tags, :broker_base_url, :basic_auth_options, :http_client, :index_entity
 
       ALL_PROVIDER_TAG_RELATION = 'pb:provider-pacts-with-tag'.freeze
       LATEST_PROVIDER_TAG_RELATION = 'pb:latest-provider-pacts-with-tag'.freeze
-      ALL_PROVIDER_RELATION = 'pb:provider-pacts'.freeze
       LATEST_PROVIDER_RELATION = 'pb:latest-provider-pacts'.freeze
       PACTS = 'pacts'.freeze
       HREF = 'href'.freeze
 
-      def initialize(provider, tags, broker_base_url, basic_auth_options, all_pacts)
+      def initialize(provider, tags, broker_base_url, basic_auth_options)
         @provider = provider
         @tags = tags
         @broker_base_url = broker_base_url
         @http_client = Pact::Hal::HttpClient.new(basic_auth_options)
-        @all_pacts = all_pacts
       end
 
-      def self.call(provider, tags, broker_base_url, basic_auth_options, all_pacts)
-        new(provider, tags, broker_base_url, basic_auth_options, all_pacts).call
+      def self.call(provider, tags, broker_base_url, basic_auth_options)
+        new(provider, tags, broker_base_url, basic_auth_options).call
       end
 
       def call
         get_index
-        if tags && tags.any?
-          get_tagged_pacts_for_provider
-        else
-          get_pacts_for_provider
-        end
+        tags&.any? ? get_tagged_pacts_for_provider : get_latest_pacts_for_provider
       end
 
       private
 
-      def get_pacts_for_provider
-        if all_pacts
-          get_all_pacts_for_provider
+      def get_tagged_pacts_for_provider
+        tags.collect do |tag|
+          link = get_link(tag)
+          tag_name = get_tag_name(tag)
+          get_pact_urls(link.expand(provider: provider, tag: tag_name).get)
+        end.flatten
+      end
+
+      def get_link(tag)
+        if tag.is_a?(String) || tag[:all] == false
+          index_entity._link(LATEST_PROVIDER_TAG_RELATION)
         else
-          get_latest_pacts_for_provider
+          index_entity._link(ALL_PROVIDER_TAG_RELATION)
         end
       end
 
-      def get_tagged_pacts_for_provider
-        if all_pacts
-          get_all_tagged_pacts_for_provider
+      def get_tag_name(tag)
+        if tag.is_a?(String)
```

Comment:
Rather than checking throughout the class if `tag.is_a?(String)`, how about in the constructor, you convert all the strings to the hash syntax, and save those to `@tags`.